services/repository/delete.go

```diff
@@ -422,3 +422,30 @@ func RemoveRepositoryFromTeam(ctx context.Context, t *organization.Team, repoID
 
 	return committer.Commit()
 }
+
+// DeleteOwnerRepositoriesDirectly calls DeleteRepositoryDirectly for all repos of the given owner
+func DeleteOwnerRepositoriesDirectly(ctx context.Context, owner *user_model.User) error {
+	for {
+		repos, _, err := repo_model.GetUserRepositories(&repo_model.SearchRepoOptions{
```

Comment:
This function needs a ctx, I think it's the reason why deadlock.

Reply:
You're right. That solved the problem.